src/quicklist.c

```diff
@@ -678,11 +787,29 @@ void quicklistDelEntry(quicklistIter *iter, quicklistEntry *entry) {
  * Returns 1 if replace happened.
  * Returns 0 if replace failed and no changes happened. */
 int quicklistReplaceAtIndex(quicklist *quicklist, long index, void *data,
-                            int sz) {
+                            size_t sz) {
     quicklistEntry entry;
     if (likely(quicklistIndex(quicklist, index, &entry))) {
+
+        int isCurrentplain = (entry.node->container == QUICKLIST_NODE_CONTAINER_NONE);
+        int isNewPlain = isLargeElement(sz);
+
+        if (isNewPlain && isCurrentplain) {
+            zfree(entry.node->entry);
+            entry.node->entry = zmalloc(sz);
```

Comment:
Would it be better to use `realloc`?

Reply:
i may be missing some context, but if we don't need the data, free+malloc is better than realloc (saves the need for memcpy)